rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java

```diff
@@ -174,21 +180,81 @@ void logUpdateCollections() {
+
+        Keccak256 rskTxHash = PegTestUtils.createHash3(1);
+
+        BtcTransaction btcTxMock = mock(BtcTransaction.class);
+        when(btcTxMock.getHash()).thenReturn(BitcoinTestUtils.createHash(1));
+        when(btcTxMock.getHashAsString()).thenReturn(rskTxHash.toHexString());
+
+        // Act
+        eventLogger.logAddSignature(federationMember, btcTxMock, rskTxHash.getBytes());
+
+        // Assert
+        commonAssertLogs(eventLogs);
+        assertTopics(3, eventLogs);
+
+        ECKey federatorECKeyFromBtcPubKey = ECKey.fromPublicOnly(federationMember.getBtcPublicKey().getPubKey());
+        String derivedRskAddress = ByteUtil.toHexString(federatorECKeyFromBtcPubKey.getAddress());
+        assertEvent(eventLogs, 0, BridgeEvents.ADD_SIGNATURE.getEvent(), new Object[]{rskTxHash.getBytes(), derivedRskAddress}, new Object[]{federatorBtcPubKey.getPubKey()});
+    }
+
+    @ParameterizedTest()
+    @MethodSource("logAddSignatureArgProvider")
+    void logAddSignature_after_rskip415(FederationMember federationMember) {
+        // Arrange
+        ActivationConfig.ForBlock arrowheadActivations = ActivationConfigsForTest.arrowhead600().forBlock(0);
+        BridgeMainNetConstants bridgeMainNetConstants = BridgeMainNetConstants.getInstance();
+        BlockTxSignatureCache signatureCache = new BlockTxSignatureCache(new ReceivedTxSignatureCache());
+        LinkedList<LogInfo> eventLogs = new LinkedList<>();
+        BridgeEventLogger eventLogger = new BridgeEventLoggerImpl(bridgeMainNetConstants, arrowheadActivations, eventLogs, signatureCache);
+
+        BtcECKey federatorBtcPubKey = federationMember.getBtcPublicKey();
+
         Keccak256 rskTxHash = PegTestUtils.createHash3(1);
-        when(btcTxMock.getHashAsString()).thenReturn("3e72fdbae7bbd103f08e876c765e3d5ba35db30ea46cb45ab52803f987ead9fb");
+
+        BtcTransaction btcTxMock = mock(BtcTransaction.class);
+        when(btcTxMock.getHash()).thenReturn(BitcoinTestUtils.createHash(1));
+        when(btcTxMock.getHashAsString()).thenReturn(rskTxHash.toHexString());
 
         // Act
-        eventLogger.logAddSignature(federatorPubKey, btcTxMock, rskTxHash.getBytes());
+        eventLogger.logAddSignature(federationMember, btcTxMock, rskTxHash.getBytes());
 
+        // Assert
         commonAssertLogs(eventLogs);
         assertTopics(3, eventLogs);
-        ECKey key = ECKey.fromPublicOnly(federatorPubKey.getPubKey());
-        String federatorRskAddress = ByteUtil.toHexString(key.getAddress());
-        assertEvent(eventLogs, 0, BridgeEvents.ADD_SIGNATURE.getEvent(), new Object[]{rskTxHash.getBytes(), federatorRskAddress}, new Object[]{federatorPubKey.getPubKey()});
+
+        String derivedRskAddress = ByteUtil.toHexString(federationMember.getRskPublicKey().getAddress());
+        assertEvent(eventLogs, 0, BridgeEvents.ADD_SIGNATURE.getEvent(), new Object[]{rskTxHash.getBytes(), derivedRskAddress}, new Object[]{federatorBtcPubKey.getPubKey()});
     }
```

Comment:
same here. We can pass the "rsk pub key" hardcoded and the activations to the args provider

Reply:
Update!